### ISSUE TITLE
New pip failure on --cache-dir

### ISSUE THREAD

**Issue by warsaw** [rtfd/readthedocs.org#1869]
It seems like the builds for mailman have been failing intermittently with a new error.  The version of pip that gets installed into the venv apparently doesn't understand --cache-dir.  This is a new failure and doesn't seem to be affected by enabling or disabling site packages.

**Comment by agjohnson** [comment]
This might be an option that disappeared in pip, though we're pinning the pip version. What version of pip was used in the virtualenv?

**Comment by rougier** [comment]
I just got the same (but random) error (https://readthedocs.org/projects/glumpy/builds/).
Sometime it works and some other time it complain about --cache-dir being an unknown option.
(And now I got another random error about not found my own .requirements while it did an hour ago)

**Comment by rougier** [comment]
I confirm the error seems to be pretty random. Sometime it works, sometime it doesn't.

**Comment by warsaw** [comment]
Any progress on this?  It still seems completely random.  Sometimes it works and sometimes it doesn't.

**Comment by agjohnson** [comment]
@rougier it looks like you resolved the issue with `.requirements` not being found, did you do anything to fix this?

@warsaw no update on this, i looked into this briefly before, but don't have a solid grasp on what is happening. It seems pip `1.5.4` is being used from the base system in some non-deterministic fashion for some python 3 projects. Have you tried wiping the versions from the project version page?

**Comment by rougier** [comment]
No, the .requirements was added pretty soon in my project but I still got random errors. However, lately, things have improved without changing anything. The build also complains from time to time about a missing .requirements while I never removed it.

**Comment by agjohnson** [comment]
I'm curious if the two are related. I don't know of any other cases of this being an issue however. Builds seem to randomly fail, even when they occur on the same server. This is going to take some deeper digging.

**Comment by joelfrederico** [comment]
I'm having this problem now. I wiped the dev branch before I built, still had the problem. It occurs quite frequently for me.

https://readthedocs.org/projects/slactrac/builds/3710671/

**Comment by WJCode** [comment]
Recently, our builds have been getting this error quite often.

https://readthedocs.org/projects/coala/builds/

I have tried to trigger our builds after wiping the stable and latest versions and that seems to work sometimes.

**Comment by agjohnson** [comment]
Thanks for the additional reports! This is still being extremely elusive. Hopefully there is some correlation to been seen here.

The interesting thing that is happening is that it is isn't happening on just one server. These errors will happen across several build servers over the course of a couple days.

Sorry there isn't more that anyone can do here, we'll try to get some priority on this and get it tracked down.

**Comment by stsewd** [comment]
Looks like the projects that are mention here has pretty solid builts (there are some failures, but none of then related to this issue). And one project hasn't had recent builds and the last 2 years.

I will close this issue in a couple of weeks if there isn't a recent report with the same issue.
